### PR TITLE
refactor: modify artifact-id logic; jsonify responses

### PR DESCRIPTION
**Pull request by jupyterjazz**
Closes https://github.com/jina-ai/finetuner.fit/issues/209 and https://github.com/jina-ai/finetuner.fit/issues/210

- Support jsonified dicts instead of `requests.Response` objects
- Change the logic behind retrieving artifact-id for downloading finetuned model(s)

### CODE REVIEW

File: finetuner/client/client.py [hunk 1]
```diff
@@ -130,18 +130,18 @@ def get_run(self, experiment_name: str, run_name: str) -> requests.Response:
...
         """List all created runs inside a given experiment.
 
         If no experiment is specified, list runs for all available experiments.
         :param experiment_name: The name of the experiment.
-        :return: List of `requests.Response` object(s).
+        :return: List of all runs.
         """
         if not experiment_name:
+            ls = self.list_experiments()
+            print(type(ls), type(ls[0]))
```
Comment (bwanglzu) [inline finetuner/client/client.py:142]: remove this one

File: finetuner/client/client.py [hunk 2]
```diff
@@ -130,18 +130,18 @@ def get_run(self, experiment_name: str, run_name: str) -> requests.Response:
...
+    def list_runs(self, experiment_name: Optional[str] = None) -> List[dict]:
         """List all created runs inside a given experiment.
 
         If no experiment is specified, list runs for all available experiments.
         :param experiment_name: The name of the experiment.
-        :return: List of `requests.Response` object(s).
+        :return: List of all runs.
         """
         if not experiment_name:
+            ls = self.list_experiments()
```
Comment (bwanglzu) [inline finetuner/client/client.py:141]:
```suggestion
            experiments = self.list_experiments()
```

**Comment by jupyterjazz** [comment]
Tests aren't passing here because we haven't updated `FINETUNER_HOST` secret on Github.

File: finetuner/client/client.py [hunk 3]
```diff
@@ -227,14 +227,19 @@ def _push_data_to_hubble(
...
         :param run_name: The name of the run.
         :param path: Directory where the model will be stored.
-        :returns: A str object indicates the download path on localhost.
+        :returns: A list of str object(s) that indicate the download path on localhost.
         """
-        artifact_id = '-'.join([self._hubble_user_id, experiment_name, run_name, MODEL])
-        response = self._hubble_client.download_artifact(id=artifact_id, path=path)
+        artifact_ids = self.get_run(experiment_name=experiment_name, run_name=run_name)[
+            ARTIFACT_IDS
+        ]
```
Comment (bwanglzu) [inline finetuner/client/client.py:240]: this part let me clarify it to you, i think it is wrong.

File: finetuner/constants.py [hunk 4]
```diff
@@ -19,3 +19,4 @@
 EVAL_DATA = 'eval_data'
 FINETUNED_MODELS_DIR = 'finetuned_models'
 MODEL = 'model'
+ARTIFACT_IDS = 'FINETUNER_ARTIFACT_IDS'
```
Comment (bwanglzu) [inline finetuner/constants.py:22]: as mentioned above, it is the env variable we send to AWS batch, not the field we saved in the database.

**Review by bwanglzu — CHANGES_REQUESTED** [review]
left some comments

File: finetuner/client/client.py (hunk 2 above)
Comment (jupyterjazz) [reply finetuner/client/client.py:141]: forgot to delete these.. :smile:

File: finetuner/constants.py (hunk 4 above)
Comment (jupyterjazz) [reply finetuner/constants.py:22]: but how do I, as a client, retrieve it then?

File: finetuner/client/base.py [hunk 5]
```diff
@@ -47,16 +47,16 @@ def _handle_request(
         method: str,
         params: Optional[dict] = None,
         json_data: Optional[dict] = None,
-    ) -> requests.Response:
+    ) -> Union[dict, List[dict]]:
```
Comment (gmastrapas) [inline finetuner/client/base.py:50]: `list[dict]`

File: finetuner/client/client.py [hunk 6]
```diff
@@ -227,14 +225,19 @@ def _push_data_to_hubble(
...
-    ) -> requests.Response:
-        """Download finetuned model from Hubble by its ID.
+    ) -> List[str]:
+        """Download finetuned model(s) from Hubble by its ID.
 
         :param experiment_name: The name of the experiment.
         :param run_name: The name of the run.
         :param path: Directory where the model will be stored.
-        :returns: A str object indicates the download path on localhost.
+        :returns: A list of str object(s) that indicate the download path on localhost.
```
Comment (gmastrapas) [inline finetuner/client/client.py:234]: localhost? You mean local disk?

File: finetuner/client/base.py (hunk 5 above)
Comment (gmastrapas) [reply finetuner/client/base.py:50]: nevermind, I guess we want to support `>=3.7`

File: finetuner/client/client.py (hunk 6 above)
Comment (jupyterjazz) [reply finetuner/client/client.py]: I honestly just copied that from hubble's `download_artifact` docstring. But I see your point, it's a bit confusing here.

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Comment by bwanglzu** [comment]
test is failing

**Comment by jupyterjazz** [comment]
@bwanglzu 
> test is failing


Tests are failing here because we haven't updated `FINETUNER_HOST` secret on Github.

**Review by gmastrapas — APPROVED** [review]
Looks good!

**Pull request merged**